mobile/lib/modules/backup/providers/backup.provider.dart

```diff
@@ -263,34 +314,46 @@ class BackupNotifier extends StateNotifier<BackUpState> {
   /// and then update the UI according to those information
   ///
   Future<void> getBackupInfo() async {
-    await Future.wait([
-      _getBackupAlbumsInfo(),
-      _updateServerInfo(),
-    ]);
-
-    await _updateBackupAssetCount();
+    final bool isEnabled = await _backgroundService.isBackgroundBackupEnabled();
+    state = state.copyWith(backgroundBackup: isEnabled);
+    if (state.backupProgress != BackUpProgressEnum.inBackground) {
+      await Future.wait([
+        _getBackupAlbumsInfo(),
+        _updateServerInfo(),
+      ]);
+
+      await _updateBackupAssetCount();
+    }
   }
 
   ///
   /// Save user selection of selected albums and excluded albums to
   /// Hive database
   ///
   void _updatePersistentAlbumsSelection() {
+    final epoch = DateTime.fromMillisecondsSinceEpoch(0, isUtc: true);
     Box<HiveBackupAlbums> backupAlbumInfoBox =
         Hive.box<HiveBackupAlbums>(hiveBackupInfoBox);
     backupAlbumInfoBox.put(
       backupInfoKey,
       HiveBackupAlbums(
         selectedAlbumIds: state.selectedBackupAlbums.map((e) => e.id).toList(),
         excludedAlbumsIds: state.excludedBackupAlbums.map((e) => e.id).toList(),
+        lastSelectedBackupTime: state.selectedBackupAlbums
+            .map((e) => e.lastBackup ?? epoch)
+            .toList(),
+        lastExcludedBackupTime: state.excludedBackupAlbums
+            .map((e) => e.lastBackup ?? epoch)
+            .toList(),
       ),
     );
   }
 
   ///
   /// Invoke backup process
   ///
-  void startBackupProcess() async {
+  Future<void> startBackupProcess() async {
+    assert(state.backupProgress == BackUpProgressEnum.idle);
```

Comment:
Would this crash the app?

Reply:
in debug mode, yes.
In release mode, no as all asserts are disabled